src/include/gurt/atomic.h

```diff
@@ -43,7 +43,13 @@
 #if HAVE_STDATOMIC
 
 #include <stdatomic.h>
+
+#ifdef __INTEL_COMPILER
+#define ATOMIC volatile
```

Comment:
(style)  Use of volatile is usually wrong: see Documentation/volatile-considered-harmful.txt